src/main/java/com/searchgears/queryanalysis/matching/DictionaryRewriterHatcher.java

```diff
@@ -17,30 +20,59 @@
  * created directly except for testing.
  * It is not a factory because it needs to go through the solr resource loader dance before it can actually
  * create the DictionaryRewriter.
+ * TODO rename to DictionaryRewriterHolder
  */
 public class DictionaryRewriterHatcher implements ResourceLoaderAware {
+    private static final String DEFAULT_SYN_FILENAME = "synonyms.txt";
```

Comment:
I don't know if there should ever be an implicit file name here. The danger would be to inadvertedly overwrite another DictionaryRewriterHatcher's temporary synonym file. 
Since this is so cleary a temporary file, I would say it should be created implicitely when the #inform comes around.

Reply:
synonyms file is now created on-the-fly when solr `inform` is triggered. Simplified method parameter passing, as well.